modular_skyrat/code/game/turfs/open/floor.dm

```diff
@@ -0,0 +1,21 @@
+/turf/open/floor
+	var/dirtiness = 0
+	var/dirtiness_updating = 0
+	var/update_dirt_steps = 15
+	var/max_dirtiness = 150
+
+//Dirtiness.
+/turf/open/floor/Entered(atom/movable/AM)
+	. = ..()
+	if(istype(AM, /mob/living))
+		dirtiness_updating++
+		if(dirtiness_updating >= update_dirt_steps)
+			update_dirtiness()
+			dirtiness_updating = 0
+
+/turf/open/floor/proc/update_dirtiness()
+	dirtiness = 0
+	for(var/obj/effect/decal/cleanable/cleanable in src)
+		dirtiness += cleanable.dirtiness
+		dirtiness += (cleanable.bloodiness/2)
+	dirtiness = min(max_dirtiness, dirtiness)
```

Comment:
Possible performance concern here.

Reply:
I'm trying to test out ways to make infections happen that make sense. This might not tax as much as bay's implementation, we'll see.